### PR TITLE
CLOUDP-108871: Delete a specific private endpoint interface

### PR DESCRIPTION
**Pull request by johansteyn**
## Proposed changes

_Jira ticket:_ CLOUDP-108871

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

I have "smoke-tested" locally this with an endpoint interface that I created previously:

```
% bin/mongocli atlas privateEndpoints gcp interfaces delete endpoint-1 --endpointServiceId 61eaca605af86411903de1dd
? Are you sure you want to delete: endpoint-1 Yes
Interface endpoint 'endpoint-1' deleted
```

### CODE REVIEW

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteInterfaceEndpoint, opts.ConfigProjectID(), provider, opts.privateEndpointServiceID, opts.privateEndpointGroupID)
+}
+
+// mongocli atlas privateEndpoint(s) gcp interface(s) delete <endpointGroupId> --endpointServiceId endpointServiceId [--projectId projectId].
```
Comment (colm-quinn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:49]: Consistency in the usage pattern:

```suggestion
// mongocli atlas privateEndpoint(s) gcp interface(s) delete <endpointGroupId> [--endpointServiceId endpointServiceId] [--projectId projectId].
```

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			if err := opts.Prompt(); err != nil {
+				return err
+			}
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().BoolVar(&opts.Confirm, flag.Force, false, usage.Force)
+	cmd.Flags().StringVar(&opts.privateEndpointServiceID, flag.EndpointServiceID, "", usage.EndpointServiceID)
+
```
Comment (colm-quinn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:73]: Redundant space?

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	return opts.Delete(opts.store.DeleteInterfaceEndpoint, opts.ConfigProjectID(), provider, opts.privateEndpointServiceID, opts.privateEndpointGroupID)
+}
+
+// mongocli atlas privateEndpoint(s) gcp interface(s) delete <endpointGroupId> --endpointServiceId endpointServiceId [--projectId projectId].
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts("Interface endpoint '%s' deleted\n", "Interface endpoint not deleted"),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <endpointGroupId>",
```
Comment (colm-quinn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:55]: [q] I see later the opt is `privateEndpointServiceID` - should we use that name everywhere?

**Review by colm-quinn — COMMENTED** [review]
Looking good, just some small nits, nothing major.

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		Aliases: []string{"rm"},
+		Short:   "Delete a specific GCP private endpoint interface for your project.",
+		Args:    require.ExactArgs(1),
```
Comment (andreaangiolillo) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:58]: you are missing `Annotations` and `Example`

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:49]: I think this is correct, we add `[` only for flags that are optional. `endpointServiceId` is required in this case

**Review by andreaangiolillo — COMMENTED** [review]
delete command is missing annotations

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 4 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:58]: you can use this as example 
https://github.com/mongodb/mongocli/blob/245f8531886aa4bd2035e25ef33f8e8c2c7f2e5b/internal/cli/atlas/privateendpoints/datalake/aws/delete.go#L56

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 1 above)
Comment (colm-quinn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:49]: Ah my bad, ignore me!

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 2 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:73]: This is a straight copy from the equivalent AWS delete code, which in turn is identical to the Azure delete code.

File: internal/cli/atlas/privateendpoints/gcp/interfaces/interfaces.go [hunk 5]
```diff
@@ -29,6 +29,7 @@ func Builder() *cobra.Command {
 		Short:   "Manage Atlas GCP private endpoint interfaces.",
 	}
 	cmd.AddCommand(DescribeBuilder())
+	cmd.AddCommand(DeleteBuilder())
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/gcp/interfaces/interfaces.go:32]:
```suggestion
	cmd.AddCommand(DescribeBuilder(), DeleteBuilder())
```

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 3 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:55]: There are two similar but distinct concepts: that of the endpoint service and that of the endpoint itself.

At the API level the DELETE method, across all cloud providers, uses path:

  /groups/{GROUP-ID}/privateEndpoint/{CLOUD-PROVIDER}/endpointService/{ENDPOINT-SERVICE-ID}/endpoint/{ENDPOINT-ID}

But in mongocli the terminology is different across cloud providers, and from my understanding of the spec doc for GCP endpoint interfaces:

  The endpointGroupId args maps to ENDPOINT-ID (the last component of the path)
  The --endpointServiceId flag maps to ENDPOINT-SERVICE-ID

So, privateEndpointServiceID in opt is not the same as endpointGroupId.

However, I notice now that privateEndpointGroupID in opt does not belong there (ie. it's redundant) so I will remove that.

File: internal/cli/atlas/privateendpoints/gcp/interfaces/interfaces.go (hunk 5 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/interfaces.go]: Thanks Gustavo - I have changed that in the latest commit.

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 4 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go:56]: Oh yes, thanks. I have added it in the latest commit.

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go (hunk 2 above)
Comment (johansteyn) [reply internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go]: But I agree: no need for redundant whitespace, so I have removed it in the latest commit.

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for those changes

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by JuliaMongo — APPROVED** [review]
LGTM!

**Pull request merged**